tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go

```diff
@@ -0,0 +1,136 @@
+			Expect(err).NotTo(HaveOccurred(), "error creating spoke pull-secret")
+
+			for host, info := range MGMTConfig.Cluster.Info.Hosts {
+				By("Create baremetalhost secret for " + host)
+				_, err = secret.NewBuilder(
+					APIClient, host, MGMTConfig.Cluster.Info.ClusterName, v1.SecretTypeOpaque).WithData(map[string][]byte{
+					"username": []byte(info.BMC.User),
+					"password": []byte(info.BMC.Password),
+				}).Create()
+				Expect(err).NotTo(HaveOccurred(), "error creating bmh secret")
+
+				By("Create baremetalhost for " + host)
+				hostBMH := bmh.NewBuilder(
+					APIClient, host, MGMTConfig.Cluster.Info.ClusterName, info.BMC.URLv4, host, info.BMC.MACAddress, "UEFI")
+				hostBMH.Definition.Spec.AutomatedCleaningMode = "disabled"
+
+				_, err = hostBMH.Create()
+				Expect(err).NotTo(HaveOccurred(), "error creating baremetalhost")
+			}
+
+			By("Create cluster deployment for IBI installation")
+			_, err = hive.NewABMClusterDeploymentBuilder(
+				APIClient, MGMTConfig.Cluster.Info.ClusterName,
+				MGMTConfig.Cluster.Info.ClusterName, MGMTConfig.Cluster.Info.ClusterName,
+				MGMTConfig.Cluster.Info.BaseDomain, MGMTConfig.Cluster.Info.ClusterName, metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"dummy": "label",
+					},
+				}).WithPullSecret(MGMTConfig.Cluster.Info.ClusterName).Create()
+			Expect(err).NotTo(HaveOccurred(), "error creating cluster deployment")
+
+			var snoNodeName string
+
+			for hostname := range MGMTConfig.Cluster.Info.Hosts {
+				snoNodeName = hostname
+
+				break
+			}
+
+			By("Create imageclusterinstall for IBI installation")
+			imageClusterInstall := ibi.NewImageClusterInstallBuilder(
+				APIClient, MGMTConfig.Cluster.Info.ClusterName, MGMTConfig.Cluster.Info.ClusterName, ibiImageSetName).
+				WithClusterDeployment(MGMTConfig.Cluster.Info.ClusterName).WithHostname(snoNodeName).
+				WithVersion(MGMTConfig.SeedClusterInfo.SeedClusterOCPVersion).
+				WithMachineNetwork(MGMTConfig.Cluster.Info.MachineCIDR.IPv4)
+
+			if MGMTConfig.PublicSSHKey != "" {
+				imageClusterInstall.WithSSHKey(MGMTConfig.PublicSSHKey)
+			}
+
+			imageClusterInstall.Definition.Spec.BareMetalHostRef = &ibiv1alpha1.BareMetalHostReference{}
+			imageClusterInstall.Definition.Spec.BareMetalHostRef.Name = snoNodeName
+			imageClusterInstall.Definition.Spec.BareMetalHostRef.Namespace = MGMTConfig.Cluster.Info.ClusterName
+			_, err = imageClusterInstall.Create()
+			Expect(err).NotTo(HaveOccurred(), "error creating imageclusterinstall")
+
+			Eventually(func() (bool, error) {
+				imageClusterInstall.Object, err = imageClusterInstall.Get()
+				if err != nil {
+					return false, err
+				}
+
+				condition, err := imageClusterInstall.GetCompletedCondition()
+				if err != nil {
+					return false, err
+				}
+
+				return condition.Status == "True" && condition.Reason == ibiv1alpha1.InstallSucceededReason, nil
+
+			}).WithTimeout(time.Minute*20).WithPolling(time.Second*5).Should(
```

Comment:
20 mins? this includes the installation of the spoke, right?

Reply:
This is for the full installation of the spoke, yes